src/endpoints/pool/pool.service.ts

```diff
@@ -0,0 +1,92 @@
+import { Injectable } from "@nestjs/common";
+import { QueryPagination } from "src/common/entities/query.pagination";
+import { GatewayService } from "src/common/gateway/gateway.service";
+import { ApiConfigService } from "src/common/api-config/api.config.service";
+import { CacheService } from "@multiversx/sdk-nestjs-cache";
+import { CacheInfo } from "src/utils/cache.info";
+import { TransactionInPool } from "src/common/gateway/entities/transaction.pool";
+import { TransactionType } from "../transactions/entities/transaction.type";
+
+@Injectable()
+export class PoolService {
+  constructor(
+    private readonly gatewayService: GatewayService,
+    private readonly apiConfigService: ApiConfigService,
+    private readonly cacheService: CacheService,
+  ) { }
+
+  async getPool(
+    pagination: QueryPagination,
+  ): Promise<TransactionInPool[]> {
+    if (!this.apiConfigService.isTransactionPoolEnabled()) {
+      return [];
+    }
+
+    const { from, size } = pagination;
+
+    const pool = await this.getEntirePool();
+    console.log(pool);
+    return pool.slice(from, from + size);
+  }
+
+  async getEntirePool(): Promise<TransactionInPool[]> {
+    return await this.cacheService.getOrSet(
+      CacheInfo.TransactionPool.key,
+      async () => await this.getTxPoolRaw(),
+      CacheInfo.TransactionPool.ttl
+    );
+  }
+
+  async getTxPoolRaw(): Promise<TransactionInPool[]> {
+    const pool = await this.gatewayService.getTransactionPool();
+    return this.parseTransactions(pool);
+  }
+
+  private parseTransactions(rawPool: any): TransactionInPool[] {
```

Comment:
What type is `rawPool`? Can we not use `any` type and use the correct type for `rawPool`?

Reply:
defined a type for it